js/renovation/utils/plugin/placeholder_item.tsx

```diff
@@ -46,8 +51,39 @@ export class PlaceholderItemProps {
 
 @Component({ defaultOptionRules: null, view: viewFunction })
 export class PlaceholderItem extends JSXComponent(PlaceholderItemProps) {
+  @Consumer(PluginsContext)
+  plugins!: Plugins;
+
+  @InternalState()
+  updateCounter = 0;
+
+  @Effect()
+  updateArgs(): () => void {
+    const disposers = this.componentDeps.map((entity) => this.plugins.watch(entity, () => {
+      this.updateCounter += 1;
+    }));
+
+    return (): void => {
+      disposers.forEach((disposer) => disposer());
+    };
+  }
+
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   get currentComponent(): any {
-    return this.props.componentTypes[this.props.index];
+    if (this.componentDeps.every((entity) => this.plugins.hasValue(entity))) {
+      return this.props.componentTypes[this.props.index];
+    }
+
+    return null;
+  }
```

Comment:
It won't render next `PlaceholderItems` if current doesn't have one of deps. Is it by design?

Reply:
It is necessary to avoid several renders on start because each dependency initialisation cause rerender without this check.